transports/email/src/main/java/org/mule/transport/email/MicrosoftExchangeImapConnector.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.transport.email;
+
+import org.mule.api.MuleContext;
+
+import java.util.Properties;
+
+import javax.mail.URLName;
+
+/**
+ * Defines a set of properties required by JavaMail in order to use IMAP with
+ * Microsoft Exchange.
+ *
+ * @since 3.8.0
+ */
+public class MicrosoftExchangeImapConnector extends ImapConnector
+{
+
+    public MicrosoftExchangeImapConnector(MuleContext context)
+    {
+        super(context);
+    }
+
+    @Override
+    protected void extendPropertiesForSession(Properties global, Properties local, URLName url)
+    {
+        super.extendPropertiesForSession(global, local, url);
+        addExchangeAuthenticationProperties(ImapConnector.IMAP, local);
+    }
+
+    /**
+     * Disables the following authentication methods in order to use LOGIN as this is the one that
+     * works with MS Exchange through IMAP.
+     * The support IMAP username format are:
+     * <ul>
+     *     <li>using email: jdoe@contoso.com</li>
+     *     <li>using Acitve Directory name (note that the name used is the one defined as 'User SamAccountName logon' in AD): CONTOSO\jdoe</li>
+     *     <li>accessing a shared mailbox (username same as previous case): CONTOSO\jdoe\sharedmailbox</li>
+     * </ul>
+     *
+     * @param protocol
+     * @param properties
+     */
+    private void addExchangeAuthenticationProperties(String protocol, Properties properties)
+    {
+        properties.put(String.format("mail.%s.auth.plain.disable",
```

Comment:
My neither, but I found several issues when trying to connect to shared mailboxes on Exchange. I would say that most of the use cases for IMAP connector with Exchange where used to connect to a user mailbox either by username or user email.
I have tried to just add these properties to the endpoint instead of using the explicit connector for MS Exchange created in this PR and that would be another way to solve this issue without adding custom IMAP connector that we would need to test with each version of MS Exchange.
```
<imaps:inbound-endpoint user="CONTOSO\jdoe\sharedmailbox" password="MyPassword" host="192.168.56.101" port="993">
            <property key="mail.imaps.auth.plain.disable" value="true"/>
            <property key="mail.imaps.auth.ntlm.disable" value="true"/>
            <property key="mail.imaps.auth.gssapi.disable" value="true"/>
        </imaps:inbound-endpoint>
```

What would you think? I know that I have implemented this but now that I found that by adding manually properties it works too I'm in favor or not adding a custom connector and close this PR without merging it.